roles/v2/pool/src/lib/job_negotiator/mod.rs

```diff
@@ -32,26 +49,76 @@ impl JobNegotiatorDownstream {
         sender.send(sv2_frame.into()).await.map_err(|_| ())?;
         Ok(())
     }
-    pub fn stay_alive(self_mutex: Arc<Mutex<Self>>) {
+    pub fn start(self_mutex: Arc<Mutex<Self>>) {
+        let recv = self_mutex.safe_lock(|s| s.receiver.clone()).unwrap();
         tokio::spawn(async move {
             loop {
-                let recv = self_mutex.safe_lock(|s| s.receiver.clone()).unwrap();
-                let _ = recv.recv().await;
+                if let Ok(message) = recv.recv().await {
+                    let mut frame: StdFrame = message.try_into().unwrap();
```

Comment:
handle unwraps by sending to the main thread. if a bad message is sent this task will fail while the others will keep running